.storybook/components/CodeViewer/CodeViewer.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { v4 as uuid } from "uuid";
+import { boolean } from "@storybook/addon-knobs";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+import { tomorrow as syntaxTheme } from "react-syntax-highlighter/dist/esm/styles/prism";
+import Button from "@paprika/button";
+import HideIcon from "@paprika/icon/lib/Times";
+import CopiedIcon from "@paprika/icon/lib/Check";
+import { getJSX, copyToClipboard } from "./CodeViewer.helpers";
+import * as sc from "./CodeViewer.styles";
+
+const propTypes = {
+  children: PropTypes.node,
+  defaultIsShown: PropTypes.bool,
+};
+
+const defaultProps = {
+  children: null,
+  defaultIsShown: false,
+};
+
+const CodeViewer = props => {
+  const { children, defaultIsShown } = props;
+
+  const [isShown, setIsShown] = React.useState(defaultIsShown);
+  const [copyLabel, setCopyLabel] = React.useState("Copy");
+  const viewerId = React.useRef(uuid()).current;
+
+  const handleToggle = () => {
+    setIsShown(prevIsShown => !prevIsShown);
+  };
+
+  const handleCopy = () => {
+    const codeSelector = `#code-viewer-${viewerId} .paprika-code-viewer pre code:last-child`;
+    copyToClipboard(codeSelector);
+    setCopyLabel(<CopiedIcon />);
+    setTimeout(() => {
+      setCopyLabel("Copy");
+    }, 1000);
```

Comment:
Why so long? :)

Reply:
🤷‍♂ 1 second is arbitrary, but it gives the user a bit of feedback that the action worked and they now have a clipboard full of code.